inst/templates/gitlab_ci/dot.gitlab-ci.yml

```diff
@@ -0,0 +1,36 @@
+variables:
+  # For caching renv library
+  RENV_PATHS_CACHE: ${CI_PROJECT_DIR}/cache
+  RENV_PATHS_LIBRARY: ${CI_PROJECT_DIR}/renv/library
+  # Cache .rhino
+  DOT_RHINO_PATH: ${CI_PROJECT_DIR}/.rhino
+  # Necessary for cypress to run.
+  # https://docs.cypress.io/guides/references/error-messages#A-Cached-Cypress-Binary-Could-not-be-found
+  # https://docs.cypress.io/guides/continuous-integration/introduction#Caching
+  CYPRESS_CACHE_FOLDER: ${CI_PROJECT_DIR}/.cache/Cypress
+
+
+default:
+  # Use deployed image with .gitlab-ci.Dockerfile
+  image: your/image:latest
+  cache:
+    # This key needs to be constant so all jobs in all branches will share the same cache
+    key: rhino-ci
+    paths:
+      - ${RENV_PATHS_CACHE}
+      - ${RENV_PATHS_LIBRARY}
+      - ${DOT_RHINO_PATH}
+      - ${CYPRESS_CACHE_FOLDER}
+
+Runs linters and tests:
+  before_script:
+    # Restores renv when cache is not available.
+    - R -e "options(renv.config.cache.symlinks = FALSE); renv::restore(clean = TRUE);"
+  script:
+    - R -e "rhino::lint_r()"
+    - R -e "rhino::lint_sass()"
+    - R -e "rhino::lint_js()"
+    - R -e "rhino::build_sass()"
+    - R -e "rhino::build_js()"
+    - R -e "rhino::test_r()"
+    - R -e "rhino::test_e2e()"
```

Comment:
I opted for a single job pipeline instead of a multi-job pipeline.

While a [multi-job pipeline looks neater](https://gitlab.com/recle.vibal/rhino-showcase-gitlab/-/pipelines/805717775), the fastest I was able to achieve with it using the docker image was about ~10mins.

[For a single job pipeline, the CI runs ~2-3 mins](https://gitlab.com/recle.vibal/rhino-showcase-gitlab/-/pipelines?page=1&scope=all&ref=feat%2Fuse-docker-image) for rhino-showcase. The pipeline page is not as informative as the multi-job pipeline because it only shows one job. If the pipeline throws an error, it is not apparent which step failed, but reading the [logs actually shows which step failed or encountered an error](https://gitlab.com/recle.vibal/rhino-showcase-gitlab/-/jobs/3935874409), so this might not be that much of a downside.

I also performed tests to ensure that the [pipeline catches errors in lint r, lint sass, lint js, build sass, and r unit tests](https://gitlab.com/recle.vibal/rhino-showcase-gitlab/-/pipelines?page=1&scope=all&ref=feat%2Fuse-docker-image). I could not test build js because I can't find a way to make the lint js pass and build js fail. For cypress, I encountered [fails when I was doing tests with cypress cache](https://gitlab.com/recle.vibal/rhino-showcase-gitlab/-/pipelines/805610689).